codex/blockexchange/engine/engine.nim

```diff
@@ -269,10 +277,6 @@ proc scheduleTasks(b: BlockExcEngine, blocks: seq[bt.Block]) {.async.} =
           break # do next peer
 
 proc resolveBlocks*(b: BlockExcEngine, blocks: seq[bt.Block]) {.async.} =
```

Comment:
Like I said above, making this slight style changes, just makes the codebase less consistent. Everyone has their own style and opinions about comments, spacing, alignment, etc - but for the sake of consistency, I would avoid making slight styling changes.

For clarity, having a comment briefly explaning what the code bellow does is better than not, we also want to prefix this with a double `##` and reserve `#` for inline comments.

Reply:
I don't like comments that add nothing and basically repeat the function name. It's just clutter. I want comments to be rare, so that when they show up they are not likely to be ignored.

Reply:
> I don't like comments that add nothing and basically repeat the function name.

I like using comments as visual queues that are there to guide you through the codebase. It is the prose version of what the code (usually, immediately bellow) does. I've found it useful when browsing an unfamiliar codebase, or reviewing code I've written a while back, specially when you lack good code exploration tooling - IDEs, etc...

I think the argument about only using comments when strictly necessary fails in practice for various reasons, but primarily because we tend to underestimate how much contextual information we're opperating with whe writting the code. What seems clear and logical when we've just written it becomes undecipherable as soon as this context is lost, usually a few days/weeks after or even worst when just starting with the codebase. For that reason, I tend to favor commenting rather than not.

The argument that "too many comments cause people to ignore them" is easily addressed by providing additional hints to the importance of the comment. Using `TODO`, `XXX`, `WARN`, etc... Usually addresses this quite cleanly.

I do agree that if the comment simply repeats the name of the function it should probably not be there, but that is a smaller annoyance overall. In the case of this particular comment at least, I think it provided some good initial context.

Overall, I'd like us to stick with prefacing the function with some context of what it's doing rather than not, but I'm open to suggestions, we should probably raise this to it's own issue/discussion and see what others think about it.